contract/test/market-actors.js

```diff
@@ -0,0 +1,467 @@
+// @ts-check
+import { E, getInterfaceOf } from '@endo/far';
+import { makePromiseKit } from '@endo/promise-kit';
+import { AmountMath } from '@agoric/ertp/src/amountMath.js';
+import { allValues, mapValues, seatLike } from './wallet-tools.js';
+
+const { entries, fromEntries, keys } = Object;
+const { Fail } = assert;
+
+/**
+ * @typedef {{
+ *   brand: Record<string, Promise<Brand>> & { timer: unknown }
+ *   issuer: Record<string, Promise<Issuer>>
+ *   instance: Record<string, Promise<Instance>>
+ *   installation: Record<string, Promise<Installation>>
+ * }} WellKnown
+ */
+
+/**
+ * @typedef {{
+ *   assetKind: Map<Brand, AssetKind>
+ * }} WellKnownKinds
+ */
+
+/**
+ * @param {import('ava').ExecutionContext} t
+ * @param {{ wallet: import('./wallet-tools.js').MockWallet }} mine
+ * @param { WellKnown & WellKnownKinds } wellKnown
+ * @param {{
+ *   rxAddr: string,
+ *   toSend: AmountKeywordRecord,
+ *   svcInstance?: Instance,
+ * }} shared
+ */
+export const payerPete = async (
+  t,
+  { wallet },
+  wellKnown,
+  { rxAddr, toSend, svcInstance },
+) => {
+  const instance = await (svcInstance || wellKnown.instance.postalSvc);
+
+  t.log('Pete offers to send to', rxAddr, 'via contract', instance);
```

Comment:
purely cosmetic: 
 do we not like template literals for sample code?

Reply:
it feels like extra allocations or something. plus, the console normally prints things nicely, where `${foo}` is likely to result in `[Object object]`.